src/alexapi/playback_handlers/soxhandler.py

```diff
@@ -0,0 +1,101 @@
+import subprocess
+from basehandler import BaseHandler, PlaybackAudioType
+
+
+MAX_VOLUME_GAIN=30
+
+
+class SoxHandler(BaseHandler):
+    def __init__(self, config, callback_report):
+        super(SoxHandler, self).__init__(config, callback_report)
+        self.__config = config
+
+        self.volume_gain = 0
+        self.media_volume_gain = 0
+
+        self.parametersSpeech = []
+        self.parametersMedia = []
+
+    def on_setup(self):
+        if self.__config['sound']['output']:
+            self.parametersSpeech.extend(['-t', self.__config['sound']['output']])
+            if self.__config['sound']['output_device']:
+                self.parametersSpeech.append(self.__config['sound']['output_device'])
+
+        if self.__config['sound']['media_output']:
+            self.parametersMedia.extend(['-t', self.__config['sound']['media_output']])
+            if self.__config['sound']['media_output_device']:
+                self.parametersMedia.append(self.__config['sound']['media_output_device'])
+        else:
+            self.parametersMedia = self.parametersSpeech
+
+        if self.__config['sound']['default_volume']:
+            self.on_set_volume(self.__config['sound']['default_volume'])
+        if self.__config['sound']['media_default_volume']:
+            self.on_set_media_volume(self.__config['sound']['media_default_volume'])
+
+        self.playback_padding = str(self.__config['sound']['playback_padding'])
+
+    def on_play(self, item):
+        # SoX can't play file URLs
+        audio_file = item.url.replace('file://', '')
+
+        sox_cmd = ['sox', '-q', audio_file]
+        if item.audio_type == PlaybackAudioType.SPEECH:
+            sox_cmd.extend(self.parametersSpeech)
+            sox_cmd.extend(['vol', str(self.volume_gain), 'dB'])
+        else:
+            sox_cmd.extend(self.parametersMedia)
+            sox_cmd.extend(['vol', str(self.media_volume_gain), 'dB'])
+
+        if item.offset:
+            sox_cmd.extend(['trim', self.__calculate_offset(item.offset)])
+
+        sox_cmd.extend(['pad', self.playback_padding, self.playback_padding])
+
+        self.report_play(' '.join(sox_cmd))
+        play_err = None
+        try:
+            p = subprocess.Popen(sox_cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            _, stderr = p.communicate()
+            if stderr:
+                play_err = stderr
+        except OSError as e:
+            play_err = e.message
+
+        if play_err:
+            self.report_error(play_err)
+        else:
+            self.report_finish()
+
+    def on_stop(self):
+        kill = subprocess.Popen(['pkill', '-KILL', 'sox'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
This looks like it kills all `sox` instances, not just ours. We can't do it like this.